### PR TITLE
Create stats initializer class skeleton.

### LINKED ISSUE #2714: Create basic stats module structure

### PR DESCRIPTION
**Pull request by pgk**
Fixes #

### Changes Proposed in this Pull Request

* A simple class skeleton, that's all.

### Testing Instructions

* Smoke-Test that nothing fatals out.

### Release Notes

* 

### New or Updated Hooks and Templates

*

### Deprecated Code

*

### Screenshot / Video

----

| Plugin build for 3fd160aee490722620d7641a8e570e8dee59324a <a href="#"></a> |
| ------------------------------------------------------------ |
| 📦 [Download plugin zip](https://wpjobmanager.com/wp-content/uploads/2024/01/wp-job-manager-zip-2713-3fd160ae.zip)                       |
| ▶️ [Open in playground](https://wpjobmanager.com/playground/?core=2024/01/2713-3fd160ae)             |

### CODE REVIEW

File: includes/class-wp-job-manager-stats.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	 *
+	 * @return WP_Job_Manager_Stats
+	 */
+	public static function instance(): WP_Job_Manager_Stats {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
```
Comment (fjorgemota) [inline includes/class-wp-job-manager-stats.php]: WDYT about using the `Singleton` trait that we already have for this? 

https://github.com/Automattic/WP-Job-Manager/blob/trunk/includes/trait-singleton.php

File: includes/class-wp-job-manager-stats.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Stats
+ *
+ * @package wp-job-manager
+ */
+
```
Comment (fjorgemota) [inline includes/class-wp-job-manager-stats.php]: Don't we want to put this into a namespace?

Not sure if we'll have more classes there, but I think it is worth asking. :)

**Review by yscik — APPROVED** [review]

**Review by fjorgemota — APPROVED** [review]
LGTM!

**Pull request merged**